daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -151,6 +154,49 @@ object TreeUtils {
   def createdReferencedCids(ev: CreatedEvent): Set[ContractId] =
     ev.createArguments.foldMap(args => args.fields.foldMap(f => valueCids(f.getValue.sum)))
 
+  /** A simple event causes the creation of a single contract and returns its contract id.
+    *
+    * A create event is a simple event. An exercise event can be a simple event.
+    */
+  case class SimpleEvent(event: TreeEvent.Kind, contractId: ContractId)
+
+  object SimpleEvent {
+    def fromTreeEvent(event: TreeEvent.Kind, tree: TransactionTree): Option[SimpleEvent] = {
+      event match {
+        case created @ Kind.Created(value) =>
+          Some(SimpleEvent(created, ContractId(value.contractId)))
+        case exercised @ Kind.Exercised(value) =>
+          val result = value.exerciseResult.flatMap {
+            _.sum match {
+              case Sum.ContractId(value) => Some(value)
+              case _ => None
+            }
+          }
+          val creates = {
+            var creates = Seq.empty[String]
```

Comment:
Maybe use a listbuffer to make this more efficient or perhaps a set (probably not very useful given that they should be unique anywhere).